### ISSUE TITLE
file will not build

### ISSUE THREAD

**Issue by SonicShadow191** [microsoft/vscode#144256]
We have written the needed data into your c
Issue Type: <b>Performance Issue</b>

Apologies if this is not how to solve the issue but I cannot build this copy of a working file since moving to another folder. I thought I had moved it safely via compressing and unzipping following advice elsewhere but if its not clear already, I have no idea what im doing wrong. Is a file path wrong? Is it because red squiggles are appearing where on the original VSC isnt having an issue at all? I notice that the build option doesnt appear in this version of the builder whilst the other loads fine.

VS Code version: Code 1.64.2 (f80445acd5a3dadef24aa209168452a3d97cc326, 2022-02-09T22:02:28.252Z)
OS version: Windows_NT x64 6.1.7601
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i3-2350M CPU @ 2.30GHz (4 x 2294)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: unavailable_off<br>opengl: enabled_on<br>rasterization: unavailable_off<br>skia_renderer: enabled_on<br>video_decode: unavailable_off<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: unavailable_off|
|Load (avg)|undefined|
|Memory (System)|7.91GB (3.64GB free)|
|Process Argv|--crash-reporter-id f26076ec-4a37-4712-b650-236bcf197aad|
|Screen Reader|yes|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    1	    96	  6592	code main
    0	    10	   888	   crashpad-handler
    0	   344	  3344	   extensionHost
    0	    30	  6584	     "C:\Users\Brett Ayers\AppData\Local\Programs\Microsoft VS Code\Code.exe" --ms-enable-electron-run-as-node "c:\Users\Brett Ayers\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\json-language-features\server\dist\node\jsonServerMain" --node-ipc --clientProcessId=3344
    0	    64	  8024	     "c:\Users\Brett Ayers\.vscode\extensions\ms-vscode.cpptools-1.9.2\bin\cpptools.exe"
    0	    26	  3380	   utility-network-service
    6	    84	  3628	   shared-process
    0	    31	  6976	     fileWatcher
    0	    38	  7928	     ptyHost
    0	     4	  4984	       winpty-process
    0	    53	  7516	         C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe
    1	    51	  6636	   issue-reporter
    0	    64	  6808	   gpu-process
    0	   193	  7140	   window (Hypercube Standard code - Visual Studio Code [Administrator])
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (Hypercube Standard code - Visual Studio Code [Administrator])
|    Folder (Hypercube Standard code): 10092 files
|      File types: svg(2457) md5(1740) map(1738) html(962) h(941) cpp(672)
|                  js(481) c(265) png(63) ino(55)
|      Conf files: launch.json(1) settings.json(1) makefile(1);
```

</details>
<details><summary>Extensions (21)</summary>

Extension|Author (truncated)|Version
---|---|---
doxdocgen|csc|1.4.0
vscode-pull-request-github|Git|0.36.2
better-cpp-syntax|jef|1.15.13
python|ms-|2022.0.1814523869
vscode-pylance|ms-|2022.2.4
jupyter|ms-|2022.1.1301854968
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.6
remote-containers|ms-|0.224.2
remote-ssh|ms-|0.74.0
remote-ssh-edit|ms-|0.74.0
remote-wsl|ms-|0.64.2
cmake-tools|ms-|1.9.2
cpptools|ms-|1.9.2
cpptools-extension-pack|ms-|1.1.0
vsliveshare|ms-|1.0.5418
vsliveshare-audio|ms-|0.1.91
vsonline|ms-|1.0.3076
platformio-ide|pla|2.4.2
python|tht|0.2.3
cmake|twx|0.0.17

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30443607
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscoreces:30445986
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30433507
vsclayoutctrc:30437038
vsrem710cf:30416617
dsvsc008:30440022
pythonvsnew555cf:30442237
vsbas813:30436447
vscscmwlcmc:30438804
helix:30440343
vsnot107cf:30443615

```

</details>

lipboard because it was too large to send. Please paste.

**Comment by roblourens** [comment]
Sorry, it's really not clear what you are asking. It sounds like this might not be an issue with vscode, but with your setup. Please give some more details.

**Comment by SonicShadow191** [comment]
Good morning Rob, thank you for getting back to me. 

You are probably right, it probably isn’t VSC itself and I’m sure I’m not using the correct terminology. I have no idea how to code beyond the basics shown to me on a 3D printer build guide. It is using the extension PlatformIO if that means anything to you. 

In short I think the problem stems from me moving the project folder containing the code for my 3D printer from the desktop to my documents which has probably ruined the file tree. 

Not knowing how to redirect it or fix the problem I then asked an ICT teacher at the school I work in to also take a look. His suggestion was to try taking a zipped copy of the files for my other machine that still worked after moving at the same time and make a new build that should change the file tree to the new folders, then change the necessary lines of code to match the smaller machine. This didn’t work either so that was when I put in the request. 

I have since found a near identical backdated  build of the Marlin firmware to the daily bug fix version I originally used and have rebuilt the code as if new. After a few problems I have managed to get VSC to successfully build the firmware again but I have not had the chance to test it yet on my machine. That is something I’m hoping to do one evening this week after work. Providing the new version does work, I can cancel the help request however, if you are able to give me any advice I’d be happy to give it a go. Learning how to move a project safely would be useful in future, especially as I’m considering building a pc next. 

Many thanks. 
Brett.
